File: server/Model/Edit.ts
import { Entity, PrimaryGeneratedColumn, Column, ManyToOne, OneToMany } from "typeorm";
import { Fandom } from "./Fandom";
import { Editor } from "./Editor";
import { Style } from "./Style";
import { Comment } from "./Comment";
import { Audio } from "./Audio";
import { Bookmark } from "./Bookmark";

@Entity()
export class Edit {
    @PrimaryGeneratedColumn()
    edit_id: number

    @Column()
    edit_pub: Date

    @ManyToOne(() => Fandom, fandom => fandom.fandom_edits)
    edit_fandom: Fandom

    @ManyToOne(() => Editor, editor => editor.editor_edits)
    edit_editor: Editor

    @ManyToOne(() => Style, style => style.style_edits)
    edit_style: Style

    @Column("json")
    edit_links: string[]

    @Column()
    edit_app: string

    @OneToMany(() => Comment, comment => comment.comment_edit)
    edit_comments: Comment[]

    @ManyToOne(() => Audio, audio => audio.audio_edits)
    edit_audio: Audio


}

File: server/src/main/java/laurriana/time2day/model/Edit.java
package laurriana.time2day.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Edit {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int editId;
    private Date editPublication;
    private String editApp;

    @ManyToOne
    private Fandom editFandom;

    @ManyToOne
    private Editor editEditor;

    @ManyToOne
    private Style editStyle;

    @ManyToOne
    private Audio editAudio;

}
